### PR TITLE
add loop and branch controls handling

### PR DESCRIPTION
**Pull request by mklkun**

### CODE REVIEW

**Review by ImenBizid — APPROVED** [review]

File: tests/test_gateway.py [hunk 1]
```diff
@@ -139,6 +139,93 @@ def test_submit_python_script_with_replicate(self):
...
+        # Creating the Continuation task
+        pythonTaskContinuation = self.gateway.createPythonTask()
+        pythonTaskContinuation.setTaskName("ContinuationPythonTask")
+        pythonTaskContinuation.setTaskImplementation("""print("Continuation task.")""")
+
+        branchScript = """
+if(1==1){
+    branch = "if";
+} else {
+    branch = "else";
```
Comment (medou-boushab) [inline tests/test_gateway.py]: do you really need this else?  1==1 evaluates to true right?

**Review by ImenBizid — CHANGES_REQUESTED** [review]
You have one hanging check

**Review by carolinepacheco — APPROVED** [review]
Ok

File: tests/test_gateway.py [hunk 2]
```diff
@@ -139,6 +139,93 @@ def test_submit_python_script_with_replicate(self):
...
+        pythonTaskLoop.setTaskName("LoopPythonTask")
+        pythonTaskLoop.setTaskImplementation("""print("Hello world!")""")
+        pythonTaskLoop.addDependency(pythonTaskStart)
+
+        loopScript = """
+if(variables.get('PA_TASK_ITERATION') < 5) {
+    loop = true;
+} else {
+    loop = false;
+}"""
```
Comment (andrewssobral) [inline tests/test_gateway.py]: I think you can put this in one single line like:
```
loopScript = "if(variables.get('PA_TASK_ITERATION') < 5) {loop = true;} else {loop = false;}"
```

File: tests/test_gateway.py [hunk 3]
```diff
@@ -139,6 +139,93 @@ def test_submit_python_script_with_replicate(self):
...
+        pythonTaskContinuation = self.gateway.createPythonTask()
+        pythonTaskContinuation.setTaskName("ContinuationPythonTask")
+        pythonTaskContinuation.setTaskImplementation("""print("Continuation task.")""")
+
+        branchScript = """
+if(1==1){
+    branch = "if";
+} else {
+    branch = "else";
+}"""
```
Comment (andrewssobral) [inline tests/test_gateway.py]: Same for this:
```
branchScript = "if(1==1) {branch = "if";} else {branch = "else";}"
```
However, it's correct to have `if(1==1)` ?

**Review by andrewssobral — CHANGES_REQUESTED** [review]

File: tests/test_gateway.py (hunk 1 above)
Comment (mklkun) [reply tests/test_gateway.py]: No. It does not change anything. It is the implementation of the branching script.

**Pull request merged**